### PR TITLE
add remaining API specs

### PR DESCRIPTION
**Pull request by mustafasoylu**
Add remaining API specs:

- Racks API
- Shelves API
- Books API

### CODE REVIEW

File: docs/api/plms.yaml [hunk 1]
```diff
@@ -285,6 +285,926 @@ paths:
...
+      operationId: updateRack
+      summary: Update Rack
+      description: Update a rack.
+      tags:
+        - Racks
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Rack'
```
Comment (dorukuzucu) [inline docs/api/plms.yaml:342]: This can contain Id and fields that will be updated instead of all Fields

File: docs/api/plms.yaml [hunk 2]
```diff
@@ -285,6 +285,926 @@ paths:
...
+                          application/json:
+                            schema:
+                              $ref: '#/components/schemas/Error'
+                      500:
+                        description: Internal Error
+                        content:
+                          application/json:
+                            schema:
+                              $ref: '#/components/schemas/Error'
+  /books/not-placed:
```
Comment (dorukuzucu) [inline docs/api/plms.yaml:1027]: should we use a query parameter here?

File: docs/api/plms.yaml [hunk 3]
```diff
@@ -285,6 +285,926 @@ paths:
...
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        500:
+          description: Internal Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /books/open-library:
```
Comment (dorukuzucu) [inline docs/api/plms.yaml:1159]: instead of having open-api in our endpoints, we should just add search

File: docs/api/plms.yaml [hunk 4]
```diff
@@ -285,6 +285,926 @@ paths:
...
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        500:
+          description: Internal Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /books/last-added:
```
Comment (dorukuzucu) [inline docs/api/plms.yaml:1071]: Should we use query parameter here instead of a seperate endpoint?

File: docs/api/plms.yaml (hunk 1 above)
Comment (mustafasoylu) [reply docs/api/plms.yaml:342]: Rack id would be required for the put method. You can check the "rack" component below.

File: docs/api/plms.yaml (hunk 2 above)
Comment (mustafasoylu) [reply docs/api/plms.yaml:1027]: This is the situation where shelf_id is empty on books. Therefore, I was unsure whether to put up an URL or query. We can have a shelfId query; when shelfId=null is sent, the API can send not placed books. Which one do you prefer?

File: docs/api/plms.yaml (hunk 4 above)
Comment (mustafasoylu) [reply docs/api/plms.yaml:1071]: with a number? such as lastAdded=10

File: docs/api/plms.yaml (hunk 3 above)
Comment (mustafasoylu) [reply docs/api/plms.yaml:1159]: will change it to /search

File: docs/api/plms.yaml (hunk 4 above)
Comment (mustafasoylu) [reply docs/api/plms.yaml:1071]: done as above

File: docs/api/plms.yaml (hunk 2 above)
Comment (dorukuzucu) [reply docs/api/plms.yaml:1027]: We could use a query parameter instead of creating a new endpoint for retrieving different types of books

**Review by dorukuzucu — APPROVED** [review]
lgtm

**Pull request merged**